### PR TITLE
Highlighted the description text on search

### PR DESCRIPTION
**Pull request by Ankit-Keshari-Vituity**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-web-react/src/app/entity/dataset/profile/schema/components/SchemaDescriptionField.tsx [hunk 1]
```diff
@@ -161,6 +169,9 @@ export default function DescriptionField({ description, onUpdate, isEdited = fal
                         }
                         suffix={EditButton}
                     >
+                        {/* <Highlight matchStyle={highlightMatchStyle} search={highlightText}>
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/dataset/profile/schema/components/SchemaDescriptionField.tsx:172]: Why is this here? Can we remove

File: datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/SchemaTable.tsx [hunk 2]
```diff
@@ -69,7 +69,7 @@ export default function SchemaTable({
     const [selectedFkFieldPath, setSelectedFkFieldPath] =
         useState<null | { fieldPath: string; constraint?: ForeignKeyConstraint | null }>(null);
 
-    const descriptionRender = useDescriptionRenderer(editableSchemaMetadata);
+    const descriptionRender = useDescriptionRenderer(editableSchemaMetadata, filterText);
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/SchemaTable.tsx:72]: Where do we actually search the fields for this text?

File: datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/utils/filterSchemaRows.ts [hunk 3]
```diff
@@ -47,14 +71,19 @@ export function filterSchemaRows(
         entityRegistry,
         formattedFilterText,
     );
+
+    const filteredEditableDescription = getDescriptionFieldPathsByMetadata(editableSchemaMetadata, formattedFilterText);
```
Comment (chriscollins3456) [inline datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/utils/filterSchemaRows.ts:75]: since we need to do slightly more work for checking things inside of `editableSchemaMetadata` and we're already doing that in `getFilteredFieldPathsByMetadata`, how do you feel about moving the edited description filter check inside of that function? that way we just have one checking for things in `editableSchemaMetadata`.

We could then just have one filter function for `matchesDescription` like you already have that is reused between that function and down below similar to how `matchesTagsOrTerms` is being used twice

**Review by chriscollins3456 — COMMENTED** [review]
nice! i have a suggestion on how to clean this up a bit, though, which should also improve filtering performance (minimize the number of times we need to loop through the list of rows)

could you also add tests to `filterSchemaRows.test.ts`?

File: datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/utils/filterSchemaRows.ts (hunk 3 above)
Comment (Ankit-Keshari-Vituity) [reply datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/utils/filterSchemaRows.ts]: Worked on that and committed the changes. Please check.

File: datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/utils/filterSchemaRows.ts [hunk 4]
```diff
@@ -11,23 +11,32 @@ function matchesTagsOrTerms(field: SchemaField, filterText: string, entityRegist
...
+            .filter((fieldInfo) => matchesTagsOrTermsOrDescription(fieldInfo, filterText, entityRegistry))
             .map((fieldInfo) => fieldInfo.fieldPath) || []
     );
 }
 
+function matchesDescription(fieldDescription: any, filterText: string) {
+    return fieldDescription?.toLocaleLowerCase().includes(filterText);
+}
+
 function matchesEditableTagsOrTerms(field: SchemaField, filteredFieldPathsByEditableMetadata: any) {
```
Comment (chriscollins3456) [inline datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/utils/filterSchemaRows.ts:32]: now that `filteredFieldPathsByEditableMetadata` is also checking by description, you can rename this function `matchesEditableTagsOrTermsOrDescription`

File: datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/utils/filterSchemaRows.ts [hunk 5]
```diff
@@ -47,14 +56,17 @@ export function filterSchemaRows(
...
     const expandedRowsFromFilter = new Set();
 
     rows.forEach((row) => {
         if (
             matchesFieldName(row.fieldPath, formattedFilterText) ||
             matchesEditableTagsOrTerms(row, filteredFieldPathsByEditableMetadata) ||
-            matchesTagsOrTerms(row, formattedFilterText, entityRegistry) // non-editable tags and terms
+            matchesEditableDescription(row, filteredFieldPathsByEditableMetadata) ||
+            matchesTagsOrTermsOrDescription(row, formattedFilterText, entityRegistry) || // non-editable tags and terms
+            matchesDescription(row.description, formattedFilterText) // non-editable description
```
Comment (chriscollins3456) [inline datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/utils/filterSchemaRows.ts:69]: since we can rename `matchesEditableTagsOrTerms` to `matchesEditableTagsOrTermsOrDescription` with the new functionality change, we don't need `matchedEditableDescription` here. also, since you're calling `matchesTagsOrTermsOrDescription` for non-editable info, you don't need `matchesDescription` either.

File: datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/utils/filterSchemaRows.ts [hunk 6]
```diff
@@ -63,7 +75,9 @@ export function filterSchemaRows(
         if (
             matchesFieldName(fieldName, formattedFilterText) ||
             matchesEditableTagsOrTerms(row, filteredFieldPathsByEditableMetadata) ||
-            matchesTagsOrTerms(row, formattedFilterText, entityRegistry)
+            matchesEditableDescription(row, filteredFieldPathsByEditableMetadata) ||
+            matchesTagsOrTermsOrDescription(row, formattedFilterText, entityRegistry) || // non-editable tags and terms
+            matchesDescription(row.description, formattedFilterText) // non-editable description
```
Comment (chriscollins3456) [inline datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/utils/filterSchemaRows.ts:80]: same thing here, we can just check `matchesFieldName`, `matchesEditableTagsOrTermsOrDescription`, and `matchesTagsOrTermsOrDescription`

File: datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/__tests__/filterSchemaRows.test.ts [hunk 7]
```diff
@@ -35,6 +39,52 @@ describe('filterSchemaRows', () => {
         expect(expandedRowsFromFilter).toMatchObject(new Set());
     });
 
+    it('should properly filter schema rows based on description', () => {
+        const filterText = 'test';
```
Comment (chriscollins3456) [inline datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/__tests__/filterSchemaRows.test.ts:43]: "test" here will actually also match on the fieldPath. can you make this more specific to test that we're only matching on `description`? so you could just change this to `const filterText = 'testing description';`

File: datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/__tests__/filterSchemaRows.test.ts [hunk 8]
```diff
@@ -35,6 +39,52 @@ describe('filterSchemaRows', () => {
...
+            testEntityRegistry,
+        );
+
+        expect(filteredRows).toMatchObject([{ fieldPath: 'testing' }]);
+        expect(expandedRowsFromFilter).toMatchObject(new Set());
+    });
+
+    it('should properly filter schema rows based on description regardless of capitalization', () => {
+        const editableSchemaMetadata = { editableSchemaFieldInfo: [] };
+        const filterText = 'TeSt';
```
Comment (chriscollins3456) [inline datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/__tests__/filterSchemaRows.test.ts:58]: same here, can you change to `'TeSting DesCriptioN'`? or to that effect

File: datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/__tests__/filterSchemaRows.test.ts [hunk 9]
```diff
@@ -35,6 +39,52 @@ describe('filterSchemaRows', () => {
...
+        expect(expandedRowsFromFilter).toMatchObject(new Set());
+    });
+
+    it('should properly filter schema rows based on editable description', () => {
+        const editableSchemaMetadata = {
+            editableSchemaFieldInfo: [
+                { fieldPath: 'customer', description: 'customer description', globalTags: null, glossaryTerms: null },
+            ],
+        };
+        const filterText = sampleTag.properties.name;
```
Comment (chriscollins3456) [inline datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/__tests__/filterSchemaRows.test.ts:76]: can you make this description in `editableSchemaFieldInfo` something different to make sure we're testing that it's matching specifically on that? so you can change it to `description: 'editable customer description'` and then make the filter text `editable customer description`

Comment (chriscollins3456) [reply datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/__tests__/filterSchemaRows.test.ts:76]: then can you add one more test that's just like this but checking that we're filtering regardless of capitalization?

**Review by chriscollins3456 — COMMENTED** [review]
nice! this is looking real close, just a few changes requested

File: datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/utils/filterSchemaRows.ts (hunk 4 above)
Comment (Ankit-Keshari-Vituity) [reply datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/utils/filterSchemaRows.ts]: Fixed and committed!!

File: datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/utils/filterSchemaRows.ts [hunk 10]
```diff
@@ -47,23 +52,28 @@ export function filterSchemaRows(
...
 
     rows.forEach((row) => {
         if (
             matchesFieldName(row.fieldPath, formattedFilterText) ||
-            matchesEditableTagsOrTerms(row, filteredFieldPathsByEditableMetadata) ||
-            matchesTagsOrTerms(row, formattedFilterText, entityRegistry) // non-editable tags and terms
+            matchesEditableTagsOrTermsOrDescription(row, filteredFieldPathsByEditableMetadata) ||
+            matchesEditableTagsOrTermsOrDescription(row, filteredFieldPathsByEditableMetadata) ||
+            matchesTagsOrTermsOrDescription(row, formattedFilterText, entityRegistry) || // non-editable tags and terms
+            matchesDescription(row.description, formattedFilterText) // non-editable description
```
Comment (chriscollins3456) [inline datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/utils/filterSchemaRows.ts:65]: so you don't need `matchesEditableTagsOrTermsOrDescription` twice, you can remove one of them.

Also you can remove `matchesDescription` since we're already checking for that in `matchesTagsOrTermsOrDescription`

File: datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/utils/filterSchemaRows.ts [hunk 11]
```diff
@@ -47,23 +52,28 @@ export function filterSchemaRows(
...
         const splitFieldPath = row.fieldPath.split('.');
         const fieldName = splitFieldPath.slice(-1)[0];
         if (
             matchesFieldName(fieldName, formattedFilterText) ||
-            matchesEditableTagsOrTerms(row, filteredFieldPathsByEditableMetadata) ||
-            matchesTagsOrTerms(row, formattedFilterText, entityRegistry)
+            matchesEditableTagsOrTermsOrDescription(row, filteredFieldPathsByEditableMetadata) ||
+            matchesEditableTagsOrTermsOrDescription(row, filteredFieldPathsByEditableMetadata) ||
+            matchesTagsOrTermsOrDescription(row, formattedFilterText, entityRegistry) || // non-editable tags and terms
+            matchesDescription(row.description, formattedFilterText) // non-editable description
```
Comment (chriscollins3456) [inline datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/utils/filterSchemaRows.ts:76]: same comment here

File: datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/__tests__/filterSchemaRows.test.ts [hunk 12]
```diff
@@ -35,6 +39,80 @@ describe('filterSchemaRows', () => {
...
+            editableSchemaFieldInfo: [
+                {
+                    fieldPath: 'customer',
+                    description: 'EdiTable CuStoMer DesCriptioN',
+                    globalTags: null,
+                    glossaryTerms: null,
+                },
+            ],
+        };
+        const filterText = 'EdiTable CuStoMer DesCriptioN';
```
Comment (chriscollins3456) [inline datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/__tests__/filterSchemaRows.test.ts:104]: isn't the point of this test to check that capitalization doesn't matter? I'd just make this all lowercase to actually test that (as opposed to the same piece of text that's in the description)

**Review by chriscollins3456 — APPROVED** [review]
nice! i have one minor comment on one of your test cases, but otherwise this is good to go
